### PR TITLE
fix StatusLine progress

### PR DESCRIPTION
**Pull request by jukzi**
Reusing a IProgressManager requires to finally call
progressMonitor.done() before using it again.

see
https://bugs.eclipse.org/bugs/show_bug.cgi?id=485748
and
https://github.com/eclipse-jdt/eclipse.jdt.ui/issues/773

### CODE REVIEW

**Comment by jukzi** [comment]
https://github.com/eclipse-platform/eclipse.platform.ui/pull/1077#issuecomment-1714948144

File: bundles/org.eclipse.ui.editors/src/org/eclipse/ui/editors/text/TextFileDocumentProvider.java [hunk 1]
```diff
@@ -557,30 +557,35 @@ protected FileInfo createFileInfo(Object element) throws CoreException {
...
-					fileBuffer= manager.getFileStoreTextFileBuffer(fileStore);
-				}
-			}
-			if (fileBuffer == null && provider != null) {
-				IPath location= provider.getPath(element);
-				if (location == null)
-					return null;
-				locationKind= LocationKind.NORMALIZE;
-				manager.connect(location, locationKind, getProgressMonitor());
+		IProgressMonitor progressMonitor= getProgressMonitor();
```
Comment (iloveeclipse) [inline bundles/org.eclipse.ui.editors/src/org/eclipse/ui/editors/text/TextFileDocumentProvider.java:560]: Can this change on TextFileDocumentProvider be in a dedicated commit with proper commit message explaining the reason?

**Review by iloveeclipse — CHANGES_REQUESTED** [review]

File: bundles/org.eclipse.ui.editors/src/org/eclipse/ui/editors/text/TextFileDocumentProvider.java (hunk 1 above)
Comment (jukzi) [reply bundles/org.eclipse.ui.editors/src/org/eclipse/ui/editors/text/TextFileDocumentProvider.java:560]: good enough?

Comment (iloveeclipse) [reply bundles/org.eclipse.ui.editors/src/org/eclipse/ui/editors/text/TextFileDocumentProvider.java:560]: > good enough?

Unfortunately github doesn't allow to comment on commit messages.

The current message says: 

> Reusing a IProgressManager requires to finally call progressMonitor.done() before using it again.

This is incorrect. Correct would be:

`Reusing a monitor received from IStatusLineManager requires to finally call progressMonitor.done() before using it again.`

Note: this info should be added to the `org.eclipse.jface.action.IStatusLineManager.getProgressMonitor()`  API.

In general this change in the IStatusLineManager is counter-intuitive, because the "modern" API like `SubMonitor` explicitly says there shouldn't be need to call `done()` at all.

I don't want to start discussion about all this monitor API mess, because it is something you haven't touched but please update commit message and IStatusLineManager javadoc.

Comment (jukzi) [reply bundles/org.eclipse.ui.editors/src/org/eclipse/ui/editors/text/TextFileDocumentProvider.java:560]: done

**Review by iloveeclipse — APPROVED** [review]
